src/test/ui/const_prop_miscompile.rs

```diff
@@ -0,0 +1,7 @@
+// run-pass
```

Comment:
Is the root `ui` folder really the best place for this test?

Reply:
We don't have an appropriate folder for this in `ui`.

Reply:
What about creating a mir-opt folder or so?

Reply:
There is a `mir` folder https://github.com/rust-lang/rust/tree/master/src/test/ui/mir

Reply:
I moved it to `mir-opt`